### PR TITLE
moving get_final_layer method functionality to GANDLF.utils to avoid

### LINKED ISSUE #406: import GANDLF.models produces circular import
**Describe the bug**

**To Reproduce**
1. initialize virtual env. with GANLDF installed
2. python
3. `>>> import GANDLF.models`
4. See error:

```python-traceback
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/edwardsb/repositories/GaNDLF/GANDLF/models/__init__.py", line 1, in <module>
    from .unet import unet, resunet
  File "/home/edwardsb/repositories/GaNDLF/GANDLF/models/unet.py", line 12, in <module>
    from .modelBase import ModelBase
  File "/home/edwardsb/repositories/GaNDLF/GANDLF/models/modelBase.py", line 8, in <module>
    from GANDLF.utils import get_linear_interpolation_mode
  File "/home/edwardsb/repositories/GaNDLF/GANDLF/utils/__init__.py", line 27, in <module>
    from .parameter_processing import (
  File "/home/edwardsb/repositories/GaNDLF/GANDLF/utils/parameter_processing.py", line 1, in <module>
    from GANDLF.models.modelBase import ModelBase
ImportError: cannot import name 'ModelBase' from partially initialized module 'GANDLF.models.modelBase' 
```
(most likely due to a circular import) (/home/edwardsb/repositories/GaNDLF/GANDLF/models/modelBase.py)

**Expected behavior**
import executes cleanly without an exception being raised

**Screenshots**

**GaNDLF Version**
0.0.14.dev0 master branch

### PR DESCRIPTION
**Pull request by brandon-edwards**
circular import issue.

Fixes #406 

## Proposed Changes
- move functionality for getting base model final layer so as to not cause circular import

## Checklist

- [ ] I have read the [`CONTRIBUTING`](https://github.com/CBICA/GaNDLF/blob/master/CONTRIBUTING.md) guide
- [ ] My PR is based from the [current GaNDLF master ](https://garygregory.wordpress.com/2016/11/10/how-to-catch-up-my-git-fork-to-master/)
- [ ] Non-breaking change (would not break existing functionality): please provide as many details as possible for any breaking change 
- [ ] Function/class source code documentation added/updated
- [ ] Code has been [blacked](https://github.com/psf/black#usage) for style consistency
- [ ] If applicable, version information [has been updated in GANDLF/version.py](https://github.com/CBICA/GaNDLF/blob/master/GANDLF/version.py)
- [ ] If adding a git submodule, add to list of exceptions for black styling in [pyproject.toml](https://github.com/CBICA/GaNDLF/blob/master/pyproject.toml) file 
- [ ] [Usage documentation](https://github.com/CBICA/GaNDLF/blob/master/docs) has been updated, if appropriate
- [ ] [History](https://github.com/CBICA/GaNDLF/blob/master/HISTORY.md) has been updated, if appropriate
- [ ] Tests added or modified to [cover the changes](https://app.codecov.io/gh/CBICA/GaNDLF); if coverage is reduced, please give explanation
- [ ] If customized dependency installation is required (i.e., a separate `pip install` step is needed for PR to be functional), please ensure it is reflected in all the files that control the CI, namely: [python-test.yml](https://github.com/CBICA/GaNDLF/blob/master/.github/workflows/python-test.yml), and all docker files [[1](https://github.com/CBICA/GaNDLF/blob/master/Dockerfile-CPU),[2](https://github.com/CBICA/GaNDLF/blob/master/Dockerfile-CUDA10.2),[3](https://github.com/CBICA/GaNDLF/blob/master/Dockerfile-CUDA11.3),[4](https://github.com/CBICA/GaNDLF/blob/master/Dockerfile-ROCm)]

### CODE REVIEW

**Review by Geeks-Sid — DISMISSED** [review]

File: GANDLF/models/__init__.py [hunk 1]
```diff
@@ -91,3 +93,35 @@
     "efficientnetb6": efficientnetB6,
     "efficientnetb7": efficientnetB7,
 }
+
+
+def get_modelbase_final_layer(final_convolution_layer):
```
Comment (sarthakpati) [inline GANDLF/models/__init__.py:98]: I think this function can be removed. Once this is done, this PR can be merged!

**Review by sarthakpati — CHANGES_REQUESTED** [review]
A function needs to be removed in `GANDLF.models.__init__`

File: GANDLF/models/__init__.py (hunk 1 above)
Comment (brandon-edwards) [reply GANDLF/models/__init__.py]: Done, thanks and sorry for missing that. I am getting a formatting complaint from git on GANDLF/models/__init__.py, but I am running black on it and it is saying nothing needs to be changed. I'm not sure how to proceed with that.

Comment (sarthakpati) [reply GANDLF/models/__init__.py]: Thank you! I am not sure what complain git is giving but black is all good, so we can happily merge! 😄

**Pull request merged**